src/newgidmap.c

```diff
@@ -46,32 +46,41 @@
  */
 const char *Prog;
 
-static bool verify_range(struct passwd *pw, struct map_range *range)
+
+static bool verify_range(struct passwd *pw, struct map_range *range, bool *allow_setgroups)
 {
 	/* An empty range is invalid */
 	if (range->count == 0)
 		return false;
 
-	/* Test /etc/subgid */
-	if (have_sub_gids(pw->pw_name, range->lower, range->count))
+	/* Test /etc/subgid. If the mapping is valid then we allow setgroups. */
+	if (have_sub_gids(pw->pw_name, range->lower, range->count)) {
+		*allow_setgroups |= true;
```

Comment:
Why are you `|=` this instead of simply assigning if it is a bool?

Reply:
I might just be dense.

Reply:
Ah, that's an artifact of my old patchset (which used a bit-mask to specify the different states rather than setting the policy in `verify_mapping`). I'll swap it to just assignment.

Reply:
Fixed, thanks!